controllers/storagecluster/exporter.go

```diff
@@ -325,7 +325,10 @@ func deployMetricsExporter(ctx context.Context, r *StorageClusterReconciler, ins
 			Spec: corev1.PodSpec{
 				Containers: []corev1.Container{
 					{
-						Args:    []string{"--namespaces", instance.Namespace},
+						Args: []string{
+							"--namespaces", instance.Namespace,
```

Comment:
This works only for the internal/external mode not for both to gether isn't it?

Reply:
It is deployed per cluster. So, at a time, exporter works on 1 cluster only (either external or internal).
`--namespace` is storagecluster namespace and `--ceph-auth-namespace` is operator namespace.
So, we can handle all combinations.